### PR TITLE
added vpn lifecycle events to webui

### PR DESCRIPTION
**Pull request by nullhook**
Resolves https://github.com/brave/brave-browser/issues/17885

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_vpn/resources/panel/vpn_panel.tsx [hunk 1]
```diff
@@ -50,7 +39,10 @@ function App () {
 
 function initialize () {
   render(<App />, document.getElementById('mountPoint'),
-  () => apiProxy.getInstance().showUI())
+  () => {
+    apiProxy.getInstance().showUI()
+    apiProxy.getInstance().createVPNConnection()
```
Comment (nullhook) [inline components/brave_vpn/resources/panel/vpn_panel.tsx:44]: @petemill Added this for now and will replace it in the future. Eventually, we're figuring out if webui should control creating/deleting vpn connection or not. 

In-order to switch regions we need to remove the current profile and add a new profile and @simonhong are discussing on how to go about controlling this.

File: components/brave_vpn/brave_vpn_service_desktop.h [hunk 2]
```diff
@@ -43,9 +43,9 @@ class BraveVpnServiceDesktop
   BraveVpnServiceDesktop(const BraveVpnServiceDesktop&) = delete;
   BraveVpnServiceDesktop& operator=(const BraveVpnServiceDesktop&) = delete;
 
-  void Connect();
-  void Disconnect();
-  void CreateVPNConnection();
+  void Connect() override;
```
Comment (simonhong) [inline components/brave_vpn/brave_vpn_service_desktop.h:46]: nit: let's put together with other existing overridden api at https://github.com/brave/brave-core/pull/9982/files#diff-af93cdebba7cdea7d991ac7d72e0d7063fbf6b9a606d3f49f733afd927546eb1L61

**Review by simonhong — COMMENTED** [review]
c++ lgtm with one nit.

File: components/brave_vpn/resources/panel/vpn_panel.tsx (hunk 1 above)
Comment (simonhong) [reply components/brave_vpn/resources/panel/vpn_panel.tsx:44]: On macOS, I found that we don't need to delete existing entry.
Instead, updating it with new region info and current connection will be disconnected if was connected.
Then, we have to ask connect again and next connection will use new vpn configuration.

File: components/brave_vpn/resources/panel/container.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import MainPanel from './components/main-panel'
+import apiProxy from './vpn_panel_api_proxy.js'
+import { ConnectionState } from './types/connection_state'
+
+function Main () {
+  const [isOn, setOn] = React.useState(false)
+  const [isSelectingRegion, setSelectingRegion] = React.useState(false)
+  const [hasError, setHasError] = React.useState(false)
+  const [status, setStatus] = React.useState<ConnectionState>(ConnectionState.DISCONNECTED)
+
```
Comment (nullhook) [inline components/brave_vpn/resources/panel/container.tsx:12]: @petemill ideally all of the states should be moved to react context. Once I test and understand how auth and fetching region list would work I can solidify this.

**Review by simonhong — APPROVED** [review]
++ for C++

**Pull request merged**